File: backend/app/api/courses.py
from fastapi import APIRouter

from app.schemas.course import CourseCreate, CourseOut, CourseStats
from app.services import course_service

router = APIRouter(prefix="/courses", tags=["courses"])


@router.get("", response_model=list[CourseOut])
def list_courses():
    return course_service.list_courses()


@router.post("", response_model=CourseOut)
def create_course(payload: CourseCreate):
    return course_service.create_course(payload.name, payload.description)


@router.get("/{course_id}", response_model=CourseOut)
def get_course(course_id: int):
    return course_service.get_course(course_id)


@router.get("/{course_id}/stats", response_model=CourseStats)
def get_course_stats(course_id: int):
    return course_service.get_course_stats(course_id)

File: frontend/src/api/courses.ts
import { apiClient } from "./client";
import type { Course, CourseStats } from "../types";

export async function listCourses(): Promise<Course[]> {
  const { data } = await apiClient.get<Course[]>("/api/courses");
  return data;
}

export async function createCourse(payload: { name: string; description?: string }): Promise<Course> {
  const { data } = await apiClient.post<Course>("/api/courses", payload);
  return data;
}

export async function getCourse(courseId: number): Promise<Course> {
  const { data } = await apiClient.get<Course>(`/api/courses/${courseId}`);
  return data;
}

export async function getCourseStats(courseId: number): Promise<CourseStats> {
  const { data } = await apiClient.get<CourseStats>(`/api/courses/${courseId}/stats`);
  return data;
}
